elyra/pipeline/processor_kfp.py

```diff
@@ -133,13 +133,20 @@ def _cc_pipeline(self, pipeline, pipeline_name):
         # Create dictionary that maps component Id to its ContainerOp instance
         notebook_ops = {}
 
-        # Preprocess the output/input artifacts
-        for pipeline_child_operation in pipeline.operations.values():
-            for dependency in pipeline_child_operation.dependencies:
-                pipeline_parent_operation = pipeline.operations[dependency]
-                if pipeline_parent_operation.outputs:
-                    pipeline_child_operation.inputs = \
-                        pipeline_child_operation.inputs + pipeline_parent_operation.outputs
+        # All previous operation outputs should be propagated troughout the pipeline
+        # In order to process this recursevily, operations input should be calculated
+        # with its parents input (which is the parent's parent output) + output
+        for pipeline_operation in pipeline.operations.values():
```

Comment:
Would you mind this as the comment (minor typos, but perhaps a little more clarity)?  Completely your call...
```
# All previous operation outputs should be propagated throughout the pipeline.
# In order to process this recursively, the current operation's inputs should be combined
# from its parent's inputs (which, themselves are derived from the outputs of their parent)
# and its parent's outputs.
```